postgres/tests/test_statements.py

```diff
@@ -665,17 +668,32 @@ def test_activity_snapshot_collection(
     check = integration_check(dbm_instance)
     check._connect()
 
-    conn = psycopg2.connect(host=HOST, dbname=dbname, user=user, password=password)
+    conn = psycopg2.connect(host=HOST, dbname=dbname, user=user, password=password, async_=1)
+    blocking_conn = psycopg2.connect(host=HOST, dbname=dbname, user="blocking_bob", password=password)
+
+    def wait(conn):
```

Comment:
Weird, why is this necessary? Haven't had to do this before.

Reply:
oh yeah, thats a good point it could be made more general purpose. The reason why I have this here is, I had to run the test query in async mode so the test wouldn't block on it completing (since it... never would). Because we are running in async mode, we have to "wait" for the connection to be established. More info here https://www.psycopg.org/docs/advanced.html#asynchronous-support

Reply:
Oh cool, I missed the `async_=1` 👍